pkg/diag/validator/pod.go

```diff
@@ -202,14 +202,18 @@ func (p *podStatus) String() string {
 }
 
 func extractErrorMessageFromWaitingContainerStatus(c v1.ContainerStatus) (proto.StatusCode, error) {
+	r := c.State.Waiting.Reason
+	if strings.HasSuffix(r, imagePullBackOff) {
+		r = imagePullBackOff
```

Comment:
Are you saying that multiple reasons could occur, or that we have "ImagePullBackOff" or "ErrImagePullBackOff"?  I'm leery with trying to transform error codes like this.  It doesn't seem hard to have both and change the case statement below for the additional condition.

Reply:
sounds good. i will add another constant for `ErrImagePullBackOff`